client/MentorPanel/src/components/input.js

```diff
@@ -62,27 +65,53 @@ const InputField = ({ text, onSelect, onChange }) => {
 class Input extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { text: '' };
+    this.state = { text: '', height: 0 };
```

Comment:
...so generally we're preferring functional components over `React.Component` but this one has to be a `React.Component` for easier access to the browser `window` var. Now it looks like we're also bringing redux `connect` back (because we want to map state to props)

Would it be possible to break out a smaller/dumb `InputComponent extends React.Component` that just uses handles presentation and then wrap it in a (functional) `Input` component that handles all the state and redux and uses hooks?

Reply:
For this case, I needed the connect because I want to be able to detect when the current question is updated so I can clear out the input field. This way I can have access to componentDidUpdate and check if the prop has changed. There are callbacks for dispatch but only for the component that is calling the dispatch. And in this case, there are several ways to send a question so that occurs in several components (inputfield, sendbutton, questions). This just seemed like the easiest way.

Reply:
ok, definitely don't want us to make the code more complicated than necessary. Something to keep an eye on though. I'd like to keep the codebase to one style as much as possible (ES6 class components vs functional components), and it seems like the general direction of `React` these days is moving towards functional